File: ios-client/AzureCommunication/AzureCommunication/Features/Calling/CallSetupViewModel.swift
import Foundation

@Observable
class CallSetupViewModel {
    let sharedState: SharedState
    private let routeManager: RouteManager

    init(sharedState: SharedState, routeManager: RouteManager) {
        self.sharedState = sharedState
        self.routeManager = routeManager
    }

    func createGroupCall() {
        routeManager.navigate(to: .calling(groupId: UUID().uuidString))
    }

    func joinGroupCall(_ groupId: String) {
        routeManager.navigate(to: .calling(groupId: groupId))
    }

    func goBack() {
        routeManager.pop()
    }
}

File: android-client/app/src/main/java/com/example/azurecommunication/features/callsetup/CallSetupViewModel.kt
package com.example.azurecommunication.features.callsetup

import androidx.lifecycle.ViewModel
import com.example.azurecommunication.shared.state.SharedState
import dagger.hilt.android.lifecycle.HiltViewModel
import java.util.UUID
import javax.inject.Inject

@HiltViewModel
class CallSetupViewModel @Inject constructor(
    private val sharedState: SharedState
) : ViewModel() {

    fun createGroupCall() {
        val groupId = UUID.randomUUID().toString()
        sharedState.setGroupId(groupId)
    }

    fun joinGroupCall(groupId: String) {
        sharedState.setGroupId(groupId)
    }

    fun goBack() {
        sharedState.leaveCall()
    }
}
